crates/ironrdp-rdpdr/src/pdu/esc.rs

```diff
@@ -1122,3 +1124,186 @@ impl rpce::HeaderlessDecode for HCardAndDispositionCall {
         Ok(Self { handle, disposition })
     }
 }
+
+/// [2.2.2.19 Transmit_Call]
+///
+/// [2.2.2.19 Transmit_Call]: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpesc/e3861cfa-e61b-4d64-b19d-f6b31e076beb
+#[derive(Debug)]
+pub struct TransmitCall {
+    pub handle: ScardHandle,
+    pub send_pci: SCardIORequest,
+    pub send_length: u32,
+    pub send_buffer: Vec<u8>,
+    pub recv_pci: Option<SCardIORequest>,
+    pub recv_buffer_is_null: bool,
+    pub recv_length: u32,
+}
+
+impl TransmitCall {
+    const NAME: &'static str = "Transmit_Call";
+
+    pub fn decode(src: &mut ReadCursor<'_>) -> PduResult<Self> {
+        Ok(rpce::Pdu::<Self>::decode(src)?.into_inner())
+    }
+}
+
+impl rpce::HeaderlessDecode for TransmitCall {
+    fn decode(src: &mut ReadCursor<'_>) -> PduResult<Self> {
+        let mut index = 0;
+        let mut handle = ScardHandle::decode_ptr(src, &mut index)?;
+        let mut send_pci = SCardIORequest::decode_ptr(src, &mut index)?;
+        ensure_size!(in: src, size: size_of::<u32>());
+        let _send_length = src.read_u32();
+        let _send_buffer_ptr = ndr::decode_ptr(src, &mut index)?;
+        let recv_pci_ptr = ndr::decode_ptr(src, &mut index)?;
+        ensure_size!(in: src, size: size_of::<u32>() * 2);
+        let recv_buffer_is_null = src.read_u32() == 1;
+        let recv_length = src.read_u32();
+
+        handle.decode_value(src)?;
+        send_pci.decode_value(src)?;
+
+        ensure_size!(in: src, size: size_of::<u32>());
+        let send_length = src.read_u32();
+        ensure_size!(in: src, size: send_length as usize);
+        let send_buffer = src.read_slice(send_length as usize).to_vec();
```

Comment:
Nit: `cast_length` should be used instead of `as usize`:
```suggestion
        let send_length: usize = cast_length!("TransmitCall", "send_buffer", src.read_u32());
        ensure_size!(in: src, size: send_length);
        let send_buffer = src.read_slice(send_length).to_vec();
```

Reply:
18b05ca972ed19e165fff7ab6066c433c905b600